src/fbas.rs

```diff
@@ -156,7 +173,10 @@ impl Fbas {
         curr_depth: u32,
         known_validators: &BTreeMap<&String, NodeIndex>,
         known_qsets: &mut BTreeMap<Qset, NodeIndex>,
+        resource_limiter: &ResourceLimiter,
     ) -> Result<NodeIndex, FbasError> {
+        resource_limiter.measure_and_enforce_limits()?;
```

Comment:
Oh, is the change that now you're _also_ threading the same resource limiter through all of the setup steps, so if you run out of memory during setup (before the first callback) you'll still be prevented from going any further?

Reply:
It turns out the resource limiter approach works well for time, but not memory. 
Since the `stop()` is checked after each internal iteration of the SAT solver, and each iteration the solver can allocate large chunks of memory based on what it's learned. The memory that exceeds the limit within the iteration can be non-negligible. 
So I went back to the global-allocator enforced memory limit approach. Which puts a hard cap over the memory usage for better safety.